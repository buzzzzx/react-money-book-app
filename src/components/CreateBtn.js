import React from "react";
import { Link } from "@reach/router";
import Ionicon from "react-ionicons";

const CreateBtn = ({ onClick }) => (
  <Link to="create" className="btn btn-primary btn-lg btn-block my-4">
    <Ionicon
      className="rounded-circle mr-2"
      fontSize="30px"
      color="#fff"
      icon="ios-add-circle-outline"
    />
    创建一条新的记账记录
  </Link>
);

export default CreateBtn;
