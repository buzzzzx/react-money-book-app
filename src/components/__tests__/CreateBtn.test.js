import React from "react";
import { shallow } from "enzyme";

import CreateBtn from "../CreateBtn";

// const props = {
//   onClick: jest.fn(),
// };

describe("test <CreateBtn />", () => {
  it("should render the component to match the snapshot", () => {
    const wrapper = shallow(<CreateBtn />);
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: need to test with Link of @reach/router
  // it("should trigger the correct callback", () => {
  //   const wrapper = shallow(<CreateBtn {...props} />);
  //   wrapper.find("button").simulate("click");
  //   expect(props.onClick).toHaveBeenCalled();
  // });
});
