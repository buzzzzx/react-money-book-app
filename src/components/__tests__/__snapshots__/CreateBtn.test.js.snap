// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <CreateBtn /> should render the component to match the snapshot 1`] = `ShallowWrapper {}`;
